cips/cip-shwap_protocol.md

```diff
@@ -0,0 +1,407 @@
+---
+cip: 18
+title: Shwap Protocol 
+description: Shwap - a new messaging framework for DA and sampling
+author: Hlib Kanunnikov (@Wondertan)
+discussions-to: https://forum.celestia.org/t/cip-shwap-protocol/1551
+status: Draft
+type: Standards Track
+category: Data Availability, Networking
+created: 2024-02-02
+---
+
+## Abstract
+
+This document specifies Shwap - the simple and expressive yet extensible messaging framework aiming to
```

Comment:
I think the name choice should be defined as early as possible

```suggestion
This document specifies Shwap (a portmanteau of share and swap)  - the simple, expressive, and extensible messaging framework aiming to
```

Reply:
agreed